sqlmesh/dbt/basemodel.py

```diff
@@ -61,6 +61,16 @@ def union(self, other: Dependencies) -> Dependencies:
 
         return dependencies
 
+    def dict(self, *args: t.Any, **kwargs: t.Any) -> t.Dict[str, t.Any]:
+        # See https://github.com/pydantic/pydantic/issues/1090
+        exclude = kwargs.pop("exclude", None) or set()
+
+        out = super().dict(*args, **kwargs, exclude={*exclude, "macros"})
+        if "macros" not in exclude:
+            out["macros"] = [macro.dict() for macro in self.macros]
+
+        return out
```

Comment:
We already have an attribute_dict method. Can that be used?

Reply:
`attribute_dict` relies on Pydantic `dict()` and fails without this change. This method overrides the Pydantic's `dict()`.

Reply:
Okay, I didn't realize it was an override.